### PR TITLE
Don't use reflection to find api package name.

### PR DESCRIPTION
**Pull request by allanca**
Also, move preference data from the old package name scheme to the new so that no data is lost on upgrades.

### CODE REVIEW

File: src/com/amplitude/api/AmplitudeClient.java [hunk 1]
```diff
@@ -779,4 +782,98 @@ private String bytesToHexString(byte[] bytes) {
...
+                    sourcePkgName = Constants.class.getPackage().getName();
+                } catch (Exception e) { }
+            }
+
+            if (targetPkgName == null) {
+                targetPkgName = Constants.PACKAGE_NAME;
+            }
+
+            // No need to copy if the source and target are the same.
+            if (Constants.PACKAGE_NAME.equals(sourcePkgName)) {
```
Comment (curtisliu) [inline src/com/amplitude/api/AmplitudeClient.java]: should this be targetPkgName.equals(sourcePkgName)?

**Comment by curtisliu** [comment]
LGTM, were you able to test the upgrade path manually on an emulator or device?

**Comment by allanca** [comment]
Yes, everything looks correct when moving between versions. I'll merge and ship this as 1.6.0

**Comment by curtisliu** [comment]
Awesome thanks!

**Pull request merged**